src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -804,7 +810,7 @@ private function checkDefinition($id, array $definition, $file)
     {
         if ($throw = $this->isLoadingInstanceof) {
             $keywords = self::$instanceofKeywords;
-        } elseif ($throw = isset($definition['resource'])) {
+        } elseif (isset($definition['resource']) || isset($definition['namespace'])) {
```

Comment:
I wouldn't add the isset checl for the namespace here, it adds confusion to me (one would then wonder how a resource can be null while namespace isn't, which contradicts the above check

Reply:
Ok, then the check for `namespace` without `resource` would have to come before this conditional... Is that acceptable?

Reply:
OK, got it, the diff mislead me. Fine as is!

Reply:
but $throw still need to be set

Reply:
Oops, yeah in master it isn't required anymore (#22750) but in 3.4 it is.

Reply:
Fixed.